cpp/src/rolling/rolling.cu

```diff
@@ -649,12 +998,35 @@ std::unique_ptr<column> rolling_window(column_view const& input,
                                        size_type min_periods,
                                        std::unique_ptr<aggregation> const& agg,
                                        rmm::mr::device_memory_resource* mr)
+{
+  return rolling_window(
+    input, empty_like(input)->view(), preceding_window, following_window, min_periods, agg, mr);
+}
+
+// Applies a fixed-size rolling window function to the values in a column.
+std::unique_ptr<column> rolling_window(column_view const& input,
+                                       column_view const& default_outputs,
+                                       size_type preceding_window,
+                                       size_type following_window,
+                                       size_type min_periods,
+                                       std::unique_ptr<aggregation> const& agg,
+                                       rmm::mr::device_memory_resource* mr)
 {
   CUDF_FUNC_RANGE();
 
   if (input.size() == 0) return empty_like(input);
   CUDF_EXPECTS((min_periods >= 0), "min_periods must be non-negative");
 
+  CUDF_EXPECTS((default_outputs.is_empty() || default_outputs.size() == input.size()),
+               "Defaults column must be either empty or have as many rows as the input column.");
+
+  CUDF_EXPECTS(
+    default_outputs.is_empty() || agg->kind == aggregation::LEAD || agg->kind == aggregation::LAG,
```

Comment:
As per my observation, this supports only fixed size windowing, in this case  we can try to handle cases where window size is 0 and other cases where ouput might be same as input, and we can just return copied version of it.

Reply:
>  we can try to handle cases where window size is 0 ... we can just return copied version of it.

#6261 actually had this. (#6261 was closed in favour of this PR.) The reason I don't have this here is that I'd need to add an `if (LEAD OR LAG` check in `rolling_window()`. If that doesn't sound ugly, I could add it back.

Reply:
I have added the short-ckt path you recommended, @rgsl888prabhu.